pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java

```diff
@@ -208,6 +209,11 @@ public BrokerResponseNative reduceOnDataTable(BrokerRequest brokerRequest,
       }
       numGroupsLimitReached |= Boolean.parseBoolean(metadata.get(MetadataKey.NUM_GROUPS_LIMIT_REACHED.getName()));
 
+      String invalidColumnNamesString = metadata.get(MetadataKey.INVALID_COLUMN_IN_QUERY.getName());
+      if (invalidColumnNamesString != null) {
+        invalidColumnNames = invalidColumnNames == null ? invalidColumnNamesString : invalidColumnNames;
+      }
```

Comment:
Can be changed as:
```
if (invalidColumnNames != null && invalidColumnNamesString != null) {
        invalidColumnNames =  invalidColumnNamesString;
      }
```
Which is a little bit more succinct. This is a minor comments, up to you to change or not.

Reply:
The if statement should be `if (invalidColumnNames == null && invalidColumnNamesString != null)`. 
Will update that in the next push. Thanks!